test/csv/parse/test_general.rb

```diff
@@ -138,28 +138,55 @@ def test_non_regex_edge_cases
     end
   end
 
-  def test_malformed_csv_cr_first_line
-    error = assert_raise(CSV::MalformedCSVError) do
-      CSV.parse_line("1,2\r,3", row_sep: "\n")
+  def test_unquoted_cr_with_lf_row_separator
+    data = "field1,field\rwith\rcr,field3\nrow2,data,here\n"
+    expected = [
+      ["field1", "field\rwith\rcr", "field3"],
+      ["row2", "data", "here"]
+    ]
+    assert_equal(expected, CSV.parse(data, row_sep: "\n"))
+  end
+
+  def test_unquoted_cr_with_custom_row_separator
+    data = "field1,field\rwith\rcr,field3|row2,data,here|"
+    expected = [
+      ["field1", "field\rwith\rcr", "field3"],
+      ["row2", "data", "here"]
+    ]
+    assert_equal(expected, CSV.parse(data, row_sep: "|"))
+  end
+
+  def test_unquoted_cr_rejected_when_included_in_row_separator
```

Comment:
We can use the same naming rule for parse error case.

```suggestion
  def test_unquoted_cr_with_crlf_row_separator
```